samples/stream_realsense.cpp

```diff
@@ -1,165 +0,0 @@
- * See the License for the specific language governing permissions and
- * limitations under the License.
- */
-#include <fcntl.h>
-#include <gst/app/gstappsrc.h>
-#include <librealsense/rs.h>
-#include <sstream>
-#include <sys/ioctl.h>
-#include <unistd.h>
-
-#include "log.h"
-#include "samples/stream_realsense.h"
-
-#define WIDTH (640)
-#define HEIGHT (480)
-#define SIZE (WIDTH * HEIGHT * 3)
-#define ONE_METER (999)
-
-struct rgb {
-    unsigned char r;
-    unsigned char g;
-    unsigned char b;
-};
-
-static void cb_need_data(GstAppSrc *appsrc, guint unused_size, gpointer user_data)
-{
-    GstFlowReturn ret;
-    rs_device *dev = (rs_device *)user_data;
-
-    rs_wait_for_frames(dev, NULL);
-    uint16_t *depth = (uint16_t *)rs_get_frame_data(dev, RS_STREAM_DEPTH, NULL);
-    struct rgb *rgb = (struct rgb *)rs_get_frame_data(dev, RS_STREAM_COLOR, NULL);
-    if (!rgb) {
-        log_error("Realsense error");
-        return;
-    }
-
-    GstBuffer *buffer
-        = gst_buffer_new_wrapped_full((GstMemoryFlags)0, rgb, SIZE, 0, SIZE, NULL, NULL);
-    for (int i = 0, end = WIDTH * HEIGHT; i < end; ++i) {
-        rgb[i].r = depth[i] * 60 / ONE_METER;
-        rgb[i].g /= 4;
-        rgb[i].b /= 4;
-    }
-
-    g_signal_emit_by_name(appsrc, "push-buffer", buffer, &ret);
-}
-
-static void cb_enough_data(GstAppSrc *src, gpointer user_data)
-{
-}
-
-static gboolean cb_seek_data(GstAppSrc *src, guint64 offset, gpointer user_data)
-{
-    return TRUE;
-}
-
-StreamRealSense::StreamRealSense()
-    : Stream()
-{
-}
-
-const std::string StreamRealSense::get_path() const
-{
-    return "/RealSense";
-}
-
-const std::string StreamRealSense::get_name() const
-{
-    return "RealSense Sample Stream";
-}
-
-const std::vector<Stream::PixelFormat> &StreamRealSense::get_formats() const
-{
-    static std::vector<Stream::PixelFormat> formats;
-    return formats;
-}
-
-GstElement *
-StreamRealSense::create_gstreamer_pipeline(std::map<std::string, std::string> &params) const
-{
-    /* librealsense */
-    rs_error *e = 0;
-    rs_context *ctx = rs_create_context(RS_API_VERSION, &e);
-    if (e) {
-        log_error("rs_error was raised when calling %s(%s): %s", rs_get_failed_function(e),
-                  rs_get_failed_args(e), rs_get_error_message(e));
-        log_error("Current librealsense api version %d", rs_get_api_version(NULL));
-        log_error("Compiled for librealsense api version %d", RS_API_VERSION);
-        return nullptr;
-    }
-    rs_device *dev = rs_get_device(ctx, 0, NULL);
-    if (!dev) {
-        log_error("Unable to access realsense device");
-        return nullptr;
-    }
-
-    /* Configure all streams to run at VGA resolution at 60 frames per second */
-    rs_enable_stream(dev, RS_STREAM_DEPTH, WIDTH, HEIGHT, RS_FORMAT_Z16, 60, NULL);
-    rs_enable_stream(dev, RS_STREAM_COLOR, WIDTH, HEIGHT, RS_FORMAT_RGB8, 60, NULL);
-    rs_start_device(dev, NULL);
```

Comment:
FPS should be configurable.  According to the recent realsense paper from Intel, the framerate has a big effect on the quality of depth data.